packages/medusa/src/migrations/1657880770264-multi_payment_cart.ts

```diff
@@ -0,0 +1,31 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class multiPaymentCart1657880770264 implements MigrationInterface {
+  name = "multiPaymentCart1657880770264"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "payment" DROP CONSTRAINT "REL_4665f17abc1e81dd58330e5854"`
+    )
+    await queryRunner.query(
+      `CREATE INDEX "IDX_aac4855eadda71aa1e4b6d7684" ON "payment" ("cart_id") WHERE canceled_at IS NOT NULL`
+    )
+    await queryRunner.query(
+      `CREATE UNIQUE INDEX "UniqPaymentSessionCartIdProviderId" ON "payment_session" ("cart_id", "provider_id")`
+    )
```

Comment:
todo: I don't think this query is related to changes in this PR - this constraint came with the payment provider typescript migration: https://github.com/medusajs/medusa/pull/1773/files#diff-e7c7c32e5cc3f7cae46795dde664d17801eda443227f979e51c3e69493004bf2